rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/core/RestControllerHelper.java

```diff
@@ -1655,4 +1700,357 @@ public ClearingRequest updateCRSize(ClearingRequest clearingRequest, Project pro
+            throw new HttpMessageNotReadableException("Index of Relationships PackageInformation invalid!");
+        }
+
+        if(CommonUtils.isNotEmpty(packageInformation.getChecksums()) &&
+                !checkIndexChecksums(packageInformation.getChecksums())) {
+            throw new HttpMessageNotReadableException("Index of Checksums PackageInformation invalid!");
+        }
+
+        packageInformation.setModerators(moderators);
+        if (!CommonUtils.isNullOrEmptyCollection(spdxDocumentActual.getSpdxPackageInfoIds())) {
+            packageInformation.setId(spdxDocumentActual.getSpdxPackageInfoIds().stream().findFirst().get());
+        }
+        return packageInformation;
+    }
+
+    public String addSPDXDocument(Release release, User user) throws TException {
+        String spdxId = "";
+        SPDXDocument spdxDocumentGenerate = generateSpdxDocument();
+        spdxDocumentGenerate.setModerators(release.getModerators());
+        spdxDocumentGenerate.setReleaseId(release.getId());
+        if (CommonUtils.isNullEmptyOrWhitespace(spdxDocumentGenerate.getId())) {
+            spdxId = spdxClient.addSPDXDocument(spdxDocumentGenerate, user).getId();
+        }
+        return spdxId;
+    }
+
+    public void addDocumentCreationInformation(String spdxId, Set<String> moderators, User user) throws TException {
+        DocumentCreationInformation documentCreationInformation = generateDocumentCreationInformation();
+        documentCreationInformation.setModerators(moderators);
+        if (isNullOrEmpty(documentCreationInformation.getSpdxDocumentId())) {
+            documentCreationInformation.setSpdxDocumentId(spdxId);
+        }
+        if (isNullOrEmpty(documentCreationInformation.getId())) {
+            documentClient.addDocumentCreationInformation(documentCreationInformation, user);
+        }
+    }
+
+    public void addPackageInformation(String spdxId, Set<String> moderators, User user) throws TException {
+        PackageInformation packageInformation = generatePackageInformation();
+        packageInformation.setModerators(moderators);
+        if (isNullOrEmpty(packageInformation.getSpdxDocumentId())) {
+            packageInformation.setSpdxDocumentId(spdxId);
+        }
+        if (isNullOrEmpty(packageInformation.getId())) {
+            packageClient.addPackageInformation(packageInformation, user);
+        }
+    }
+
+    public String addSPDX(Release release, User user) throws TException {
+        String spdxId = addSPDXDocument(release, user);
+        if (CommonUtils.isNotNullEmptyOrWhitespace(spdxId)) {
+            throw new HttpMessageNotReadableException("Add SPDXDocument Failed!");
+        }
+        addDocumentCreationInformation(spdxId, release.getModerators(), user);
+        addPackageInformation(spdxId, release.getModerators(), user);
+        return spdxId;
+    }
+
+    public boolean checkIndexCreator(Set<Creator> creators) {
```

Comment:
Adding all these validation functions to `RestControllerHelper` does not seem right to me. But I cannot think of any other better place to put them ATM.